python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -216,17 +220,62 @@ def train_step(self, graph, feat, tmp, **kwargs):
         Tensor
             A scalar tensor representing the loss.
         """
+        assert (
+            self.graph_explanation
+        ), '"graph_explanation" must be True to call this method.'
+
         self.model = self.model.to(graph.device)
         self.elayers = self.elayers.to(graph.device)
 
-        pred = self.model(graph, feat, embed=False, **kwargs).argmax(-1).data
+        pred = self.model(graph, feat, embed=False, **kwargs)
+        pred = pred.argmax(-1).data
 
         prob, _ = self.explain_graph(
             graph, feat, tmp=tmp, training=True, **kwargs
         )
 
-        loss_tmp = self.loss(prob, pred)
-        return loss_tmp
+        loss = self.loss(prob, pred)
+        return loss
+
+    def train_step_node(self, node_id, graph, feat, tmp, **kwargs):
+        r"""Compute the loss of the explanation network
+
+        Parameters
+        ----------
+        node_id : int
+            The ID of the node to target for training.
```

Comment:
Does your implementation support multiple node IDs? Does PGExplainer also support multi-instance explanation for node-level predictions?

Reply:
A node classification model is typically optimized to output the right predictions for multiple nodes simultaneously in a single training iteration. So should a multi-instance explainer for node-predictions. I expect the implementation to be compatible with multiple nodes for a training iteration. This also needs to be documented in the function doc. What do you think?

Reply:
I think I understand, for node-classification It should be using the node losses across the entire graph rather than targeting the input node. We can perform this over all input nodes for a `train_step_nodes`; would you expect this kind parameter?

```
    nodes : int, iterable[int], tensor
        The nodes from the graph used to train explanations, which cannot
        have any duplicate value. 
```

Reply:
Yes.